presto-common/src/main/java/com/facebook/presto/common/block/ArrayBlockBuilder.java

```diff
@@ -292,6 +292,14 @@ public BlockBuilder newBlockBuilderLike(BlockBuilderStatus blockBuilderStatus)
         return new ArrayBlockBuilder(blockBuilderStatus, values.newBlockBuilderLike(blockBuilderStatus), newSize);
     }
 
+    @Override
+    public BlockBuilder newBlockBuilderLike(BlockBuilderStatus blockBuilderStatus, int expectedEntries)
+    {
+        int newSize = max(calculateBlockResetSize(positionCount), expectedEntries);
+        int valueExpectedEntries = positionCount == 0 ? expectedEntries : (int) (((long) offsets[positionCount] * expectedEntries) / positionCount);
```

Comment:
can this lead to int overflow?

Reply:
Yes it can. `offsets[positionCount]` could be millions and `expectedEntries` could be thousands. The following code
```
        int offset = 2000000;
        int expectedEntries = 5000;
        int positionCount = 1000;
        int result1 = offset * expectedEntries / positionCount;
        int result2 = (int) ((long)offset * expectedEntries / positionCount);
        System.out.println( result1 + " " + result2 );
```
outputs
```
1410065 10000000
```